pokedex/pokemon.lua

```diff
@@ -889,10 +890,20 @@ local function get_damage_mod_stab(pokemon, move)
 		end
 	end
 
-	-- Figure out the STAB
+	-- Figure out the STAB and Trainer Pokemon Type Damage
 	if is_attack then
 		trainer_stab = trainer.get_all_levels_STAB()
 		for _, t in pairs(M.get_type(pokemon)) do
+			-- Figure out the higest value of the "pokemon_type_damage_bonus
+			local type_poke_dmg = trainer.get_pokemon_type_damage_bonus(t)
+			if type_poke_dmg ~= 0 then
+				if trainer_pokemon_type_damage == nil then
+					trainer_pokemon_type_damage = type_poke_dmg
+				else
+					trainer_pokemon_type_damage = math.max(trainer_pokemon_type_damage, type_poke_dmg)
+				end
+			end
```

Comment:
Not super delighted with this way of doing it. Can you think of a better way?
What I want to achieve is to add the highest of the values, but it can also be minus values. So any value other than 0 is relevant to us.